src/EFCore.Relational/Storage/RelationalTypeMapping.cs

```
@@ -259,7 +275,9 @@ public virtual string GenerateSqlLiteral([CanBeNull]object value)
                 return "NULL";
             }
 
-            return GenerateSqlLiteralValue((dynamic)value);
+            return "Unknown type " + value.GetType() +
```

Can this just be something like `string.Format(CultureInfo.InvariantCulture, "{0}", value);`? If not, should throw an exception? Or be an abstract method maybe using the template method pattern?